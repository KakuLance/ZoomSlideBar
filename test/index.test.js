import React from 'react'
import _ from 'lodash'

import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import ZoomSlideBar from '../src'

Enzyme.configure({ adapter: new Adapter() })

describe('A ZoomSlideBar with only required props should', () => {
  const low = 0
  const high = 100
  const max = 100
  const min = 0
  const step = 1

  let slideBar
  let onChangeLow
  let onChangeHigh
  let onChangePos
  let rootDiv

  beforeEach(() => {
    onChangeLow = jest.fn()
    onChangeHigh = jest.fn()
    onChangePos = jest.fn()

    rootDiv = document.createElement('div')
    rootDiv.id = 'react-root'
    rootDiv.style.width = '800px'
    document.body.appendChild(rootDiv)

    slideBar = Enzyme.mount(
      <ZoomSlideBar
        Low={low}
        High={high}
        Max={max}
        Min={min}
        onChangeLow={onChangeLow}
        onChangeHigh={onChangeHigh}
        onChangePos={onChangePos} />,
        { attachTo: rootDiv }
    )
  })

  afterEach(() => {
    rootDiv.remove()
  })

  it('matches snapshot', () => {
    expect(slideBar).toMatchSnapshot()
  })

  it('requires all necessary props', () => {
    expect(slideBar.props().Low).toBeDefined()
    expect(slideBar.props().High).toBeDefined()
    expect(slideBar.props().Max).toBeDefined()
    expect(slideBar.props().Min).toBeDefined()
    expect(slideBar.props().onChangeLow).toBeDefined()
    expect(slideBar.props().onChangeHigh).toBeDefined()
    expect(slideBar.props().onChangePos).toBeDefined()
  })

  it('renders slider, low position drag block and high position drag block', () => {
    expect(slideBar.instance().slider).toBeDefined()
    expect(slideBar.instance().Low).toBeDefined()
    expect(slideBar.instance().High).toBeDefined()
  })

  it('correctly initialises all undefined prop values', () => {
    expect(slideBar.props().Step).toBe(1)
    expect(slideBar.props().Decimal).toBe(0)
    expect(slideBar.props().BackUnit).toBe('')
    expect(slideBar.props().Vertical).toBe(false)
    expect(slideBar.props().DisplayLabel).toBe(false)
  })

  it('correctly initialises ZoomSlideBar state after mount', () => {
    // Low pos object
    expect(slideBar.state('LowPos').Value).toBe(low)
    expect(slideBar.state('LowPos').Label).toBe(`${low}`)
    expect(slideBar.state('LowPos').Position).toBe(0)
    expect(slideBar.state('LowPos').Percentage).toBe(0)

    // High pos object
    expect(slideBar.state('HighPos').Value).toBe(high)
    expect(slideBar.state('HighPos').Label).toBe(`${high}`)
    expect(slideBar.state('HighPos').Position).toBe(800)
    expect(slideBar.state('HighPos').Percentage).toBe(1)

    // Other state parameters
    const stepRange = _.range(min, max + step, step)

    expect(slideBar.state('dragging')).toBe(null)
    expect(slideBar.state('Step')).toBe(1)
    expect(slideBar.state('Steps').length).toBe(stepRange.length)
    expect(slideBar.state('StepLength')).toBe(8)
    expect(slideBar.state('dragOffset')).toBe(0)
    expect(slideBar.state('startPos')).toBe(0)
    expect(slideBar.state('eventType')).toBe(null)
  })
})
