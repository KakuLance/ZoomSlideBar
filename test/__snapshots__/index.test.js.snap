// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A ZoomSlideBar with only required props should matches snapshot 1`] = `
.c0 {
  position: absolute;
  top: 0px;
  right: 14px;
  left: 14px;
}

.c1 {
  opacity: 0.3;
  width: 14px;
  height: 14px;
  border: 2px solid #8B949C;
  background: #8B949C;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transform: translateY(50%);
  -ms-transform: translateY(50%);
  transform: translateY(50%);
  position: absolute;
  bottom: 0;
  right: NaN%;
}

.c3 {
  opacity: 0.3;
  width: 14px;
  height: 14px;
  border: 2px solid #8B949C;
  background: #8B949C;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transform: translateY(50%);
  -ms-transform: translateY(50%);
  transform: translateY(50%);
  position: absolute;
  bottom: 0;
  left: NaN%;
}

.c2 {
  opacity: 0.3;
  background: #CDCDD5;
  cursor: pointer;
  -webkit-transform: translateY(50%);
  -ms-transform: translateY(50%);
  transform: translateY(50%);
  position: absolute;
  height: 8px;
  bottom: 0;
  left: NaN%;
  right: NaN%;
}

<ZoomSlideBar
  BackUnit=""
  Decimal={0}
  DisplayLabel={false}
  High={100}
  Low={0}
  Max={100}
  Min={0}
  Step={1}
  Vertical={false}
  onChangeHigh={[MockFunction]}
  onChangeLow={[MockFunction]}
  onChangePos={[MockFunction]}
>
  <style__Root>
    <div
      className=""
    >
      <style__Bar
        innerRef={[Function]}
        vertical={false}
      >
        <div
          className="c0"
        >
          <div>
            <style__Dot
              innerRef={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              posStyle="right: NaN%;"
              vertical={false}
            >
              <div
                className="c1"
                onMouseDown={[Function]}
                onTouchStart={[Function]}
              />
            </style__Dot>
          </div>
          <style__PosBar
            innerRef={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            posStyle="
          left: NaN%;
          right: NaN%;
        "
            vertical={false}
          >
            <div
              className="c2"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </style__PosBar>
          <div>
            <style__Dot
              innerRef={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              posStyle="left: NaN%;"
              vertical={false}
            >
              <div
                className="c3"
                onMouseDown={[Function]}
                onTouchStart={[Function]}
              />
            </style__Dot>
          </div>
        </div>
      </style__Bar>
    </div>
  </style__Root>
</ZoomSlideBar>
`;
